### PR TITLE
[API Proposal]: add ToolStripItem.SelectedChanged event #8548

### PR DESCRIPTION
**Pull request by Epica3055**
related to [PR_8409](https://github.com/dotnet/winforms/pull/8409), [Issue_8548](https://github.com/dotnet/winforms/issues/8548)

Since the author of this [PR_8409](https://github.com/dotnet/winforms/pull/8409) seems to unable to continue the PR, so create a new PR here

### CODE REVIEW

**Comment by ricardobossan** [comment]
@Epica3055 LGTM but needs to solve conflicts.

**Comment by lonitra** [comment]
Can you also add tests to test the new API?

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemTests.cs [hunk 1]
```diff
@@ -15398,6 +15398,53 @@ public void ToolStripItem_IDropTargetOnDragOver_Invoke_CallsDragOver(DragEventAr
         Assert.Equal(1, callCount);
     }
 
+    [WinFormsFact]
+    public void ToolStripItem_OnItemSelected()
+    {
+        using var menuStrip1 = new MyMenuStrip();
+        using var toolStripMenuItem1 = new ToolStripMenuItem();
+        using var toolStripMenuItem2 = new ToolStripMenuItem();
+        using var toolStripMenuItem3 = new ToolStripMenuItem();
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemTests.cs:15407]: Please use target type new:
```suggestion
        using MyMenuStrip menuStrip1 = new();
        using ToolStripMenuItem toolStripMenuItem1 = new();
        using ToolStripMenuItem toolStripMenuItem2 = new();
        using ToolStripMenuItem toolStripMenuItem3 = new();
```

**Review by lonitra — COMMENTED** [review]
Small comment otherwise looks good to me. Since this is a new API, this will need to be documented. Could you open a [docs issue](https://github.com/dotnet/docs/issues) to get this documented?

**Comment by Epica3055** [comment]
@lonitra       [here](https://github.com/dotnet/docs/issues/38716) is the doc.

**Review by lonitra — APPROVED** [review]
LGTM! 👍

**Comment by Zheng-Li01** [comment]
Verified the issue on .NET 9.0.100-alpha.1.23614.10 with dlls build form main branch.
The SelectedChanged event has been added for toolStripMenuItem1, however the error appeared after double clicked the event.
But the error not appeared when testing it on main repo. Will verify it again when getting real 9.0 SDK.

image

**Pull request merged**